extensions/gitpod-web/src/extension.ts

```diff
@@ -399,86 +506,181 @@ export class GitpodWorkspaceTreeDataProvider implements vscode.TreeDataProvider<
-				const prevStatus = port.status;
+				const port = this.ports.ports.get(localPort) || new PortTreeItem(new GitpodWorkspacePort(localPort, this.context, portStatus, tunnel));
+				// const prevStatus = port.port.status;
 				this.ports.ports.set(localPort, port);
 
-				port.status = currentStatus;
-				port.tunnel = this.tunnels.get(localPort);
-				port.label = name ? `${name}: ${localPort}` : `${localPort}`;
-				if (description) {
-					port.tooltip = name ? `${name} - ${description}` : description;
-				}
-				if (port.remotePort && port.remotePort !== localPort) {
-					port.label += ':' + port.remotePort;
-				}
+				port.port.update(portStatus, tunnel);
 
-				const accessible = exposed || port.tunnel;
-
-				// We use .public here because https://github.com/gitpod-io/openvscode-server/pull/360#discussion_r882953586
-				const isPortTunnelPublic = !!port.tunnel?.public;
-				if (!served) {
-					port.description = 'not served';
-					port.iconPath = new vscode.ThemeIcon('circle-outline');
-				} else if (!accessible) {
-					if (portStatus.getAutoExposure() === PortAutoExposure.FAILED) {
-						port.description = 'failed to expose';
-						port.iconPath = new vscode.ThemeIcon('warning', new vscode.ThemeColor('editorWarning.foreground'));
-					} else {
-						port.description = 'detecting...';
-						port.iconPath = new vscode.ThemeIcon('circle-filled', new vscode.ThemeColor('editorWarning.foreground'));
-					}
-				} else {
-					port.description = 'open';
-					if (port.tunnel) {
-						port.description += ` on ${isPortTunnelPublic ? 'all interfaces' : 'localhost'}`;
-					}
-					if (exposed) {
-						port.description += ` ${exposed.visibility === PortVisibility.PUBLIC ? '(public)' : '(private)'}`;
-					}
-					port.iconPath = new vscode.ThemeIcon('circle-filled', new vscode.ThemeColor('ports.iconRunningProcessForeground'));
-				}
+				port.label = port.port.info.label;
+				port.tooltip = port.port.info.tooltip;
+				port.description = port.port.info.description;
+				port.iconPath = port.port.info.iconPath;
+				port.contextValue = port.port.info.contextValue;
 
-				port.contextValue = 'port';
-				if (served) {
-					port.contextValue = 'served-' + port.contextValue;
-				}
-				if (exposed) {
-					port.contextValue = 'exposed-' + port.contextValue;
-					port.contextValue = (exposed.visibility === PortVisibility.PUBLIC ? 'public-' : 'private-') + port.contextValue;
-				}
-				if (port.tunnel) {
-					port.contextValue = (isPortTunnelPublic ? 'network-' : 'host-') + port.contextValue;
-				}
-				if (!accessible && portStatus.getAutoExposure() === PortAutoExposure.FAILED) {
-					port.contextValue = 'failed-' + port.contextValue;
-				}
-				if (isExposedServedGitpodWorkspacePort(port) && !isExposedServedPort(prevStatus)) {
-					this.onDidExposeServedPortEmitter.fire(port);
-				}
+				// if (isExposedServedGitpodWorkspacePort(port.port) && !isExposedServedPort(prevStatus)) {
```

Comment:
why is it commented out. did we move it somewhere else or some logic was changed?

Reply:
Yeah, we will handle this only for one place, any of them is ok since we will not DEPRECATE treeView